code/datums/quirks/negative.dm

```diff
@@ -982,3 +982,134 @@
 	medical_record_text = "Patient is not literate."
 	hardcore_value = 8
 	mail_goodies = list(/obj/item/pai_card) // can read things for you
+
+/datum/quirk/old_man_jenkins
+	name = "Old Man Jenkins"
+	desc = "You require a cane like object in your hand for mobility."
+	icon = "person-cane"
+	value = -4
+	hardcore_value = 4
+	gain_text = "<span class='danger'>You feel old!</span>"
+	lose_text = "<span class='notice'>You feel young again.</span>"
+	medical_record_text = "Patient suffers from a lack of mobility without a cane as support."
+	mail_goodies = list(/obj/item/cane)
+	var/datum/weakref/left_cane
+	var/datum/weakref/right_cane
+
+/datum/quirk/old_man_jenkins/add_unique()
+	var/mob/living/carbon/human/human_holder = quirk_holder
+
+	var/turf/holder_turf = get_turf(human_holder)
+	var/obj/item/cane/spawn_cane = new /obj/item/cane(holder_turf)
+	human_holder.put_in_hands(spawn_cane)
+
+	var/obj/item/left_hand_item = human_holder.held_items[1]
+	var/obj/item/right_hand_item = human_holder.held_items[2]
+	var/is_cane_left_hand = istype(left_hand_item) && HAS_TRAIT(left_hand_item, TRAIT_CANE_TOOL)
+	var/is_cane_right_hand = istype(right_hand_item) && HAS_TRAIT(right_hand_item, TRAIT_CANE_TOOL)
+	var/is_holding_cane = is_cane_left_hand || is_cane_right_hand
+
+	if(!is_holding_cane)
+		human_holder.add_movespeed_modifier(/datum/movespeed_modifier/bad_back_slowdown) // TODO make this it's own modifier at some point
+		quirk_holder.AddElement(/datum/element/waddling)
+	else if(is_holding_cane)
+		if(is_cane_left_hand)
+			left_cane = WEAKREF(left_hand_item)
+			RegisterSignal(left_hand_item, COMSIG_ITEM_DROPPED, .proc/on_unequipped_cane)
+			RegisterSignal(left_hand_item, COMSIG_PARENT_QDELETING, .proc/on_unequipped_cane)
+			RegisterSignal(left_hand_item, COMSIG_MOVABLE_MOVED, .proc/on_unequipped_cane)
+		if(is_cane_right_hand)
+			right_cane = WEAKREF(right_hand_item)
+			RegisterSignal(right_hand_item, COMSIG_ITEM_DROPPED, .proc/on_unequipped_cane)
+			RegisterSignal(right_hand_item, COMSIG_PARENT_QDELETING, .proc/on_unequipped_cane)
+			RegisterSignal(right_hand_item, COMSIG_MOVABLE_MOVED, .proc/on_unequipped_cane)
```

Comment:
Change these to `RegisterSignals`

Reply:
Applied with 90a12b56041140e27523e06b21359333cd038634